### PR TITLE
Benchmarks v2 merge

### PR DESCRIPTION
**Pull request by magicmilo**
This is a combination of branches:
- feat/reintroduceV2Bencmarks
- getAddressAPI-fixes

#### Work done
- Added migrations
-- Added getAddressAPI fields to postcodes table
-- Added CssrID to Establishment
-- Add uuid to postcodes table
-- Added Update CssrID migration complete match
- Changed backend test reporter
- Removed 'update' hook that queried getAddressAPI
- Added methods to Cssr
- Added methods to pcodedata
- Added methods to postcodes as fallback queries and caches getAddressAPI
- Fixed broken tests
- Added tests

- Added v2 endpoint for benchmarks and rankings
-- Reintroduced V2 functionality to new endpoints
- Added benchmarksServiceBase class to front end
-- Added benchmarksV2Service wrapper to front end, extends new base class
-- Updated existing benchmarksService wrapper to extend the new base class
-- Added benchmarksServiceFactory to determine which service wrapper to use

#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: migrations/20231020115826-associate-cssrids-to-establishment.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+module.exports = {
+  async up(queryInterface) {
+    return queryInterface.sequelize.query(
+      ' \
+      UPDATE cqc."Establishment" \
+      SET "CssrID" = cssr."CssrID" \
+      FROM cqcref.pcodedata pcd \
+      JOIN cqc."Cssr" cssr ON pcd.local_custodian_code = cssr."LocalCustodianCode" \
+      WHERE cqc."Establishment"."PostCode" = pcd.postcode;',
+    );
```
Comment (Zuhal-Ayob) [inline migrations/20231020115826-associate-cssrids-to-establishment.js:13]: Hi, any reason for this?

File: package.json [hunk 2]
```diff
@@ -111,7 +111,7 @@
     "request": "^2.88.2",
     "rxjs": "^6.6.7",
     "rxjs-compat": "^6.6.7",
-    "sequelize": "^6.29.0",
+    "sequelize": "^6.33.0",
     "sequelize-cli": "^6.6.0",
```
Comment (Zuhal-Ayob) [inline package.json:115]: is there any breaking change in this new version of sequelize ?

File: migrations/20231020115826-associate-cssrids-to-establishment.js (hunk 1 above)
Comment (Jonopono123) [reply migrations/20231020115826-associate-cssrids-to-establishment.js:13]: Now that we've added a Cssr column to the establishments table, we need to populate it with the correct data. This update will get the cssrId from the cssr table, and update each row in the establishment table accordingly.

File: server/models/BulkImport/csv/workplaceCSVValidator.js [hunk 3]
```diff
@@ -2786,11 +2786,14 @@ class WorkplaceCSVValidator {
...
-      const workerFromCSV = myJSONWorkers.find((w) => {
+
+    const establishmentWorkers = await Establishment.fetchMyEstablishmentsWorkers(this.id, this._key);
+
+    establishmentWorkers.forEach((worker) => {
+      let workerFromCSV = myJSONWorkers.find((w) => {
         return w.uniqueWorkerId === worker.uniqueWorker;
       });
-      if (workerFromCSV && dataInDB.includes(workerFromCSV._status)) {
+      if (workerFromCSV && dataInDB.includes(workerFromCSV.status)) {
```
Comment (Zuhal-Ayob) [inline server/models/BulkImport/csv/workplaceCSVValidator.js:2796]: Any reason for this @magicmilo ?

File: server/test/unit/routes/training.spec.js [hunk 4]
```diff
@@ -111,44 +111,45 @@ describe('training route', () => {
         );
       });
```
Comment (Zuhal-Ayob) [inline server/test/unit/routes/training.spec.js:113]: Why these tests are commented out?

File: src/app/features/new-dashboard/dashboard/dashboard.component.ts [hunk 5]
```diff
@@ -62,16 +62,15 @@ export class NewDashboardComponent implements OnInit, OnDestroy {
       this.canViewListOfWorkers && this.setWorkersAndTrainingValues();
     }
 
-      // this.subscriptions.add(
```
Comment (Zuhal-Ayob) [inline src/app/features/new-dashboard/dashboard/dashboard.component.ts:65]: @Jonopono123 delete the commented out code

**Review by Zuhal-Ayob — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (magicmilo) [reply package.json:115]: No this isn't a major version change.

File: server/models/BulkImport/csv/workplaceCSVValidator.js (hunk 3 above)
Comment (magicmilo) [reply server/models/BulkImport/csv/workplaceCSVValidator.js:2796]: During testing, at this point _status is not initialised but status is. Going to look at the rest of the file to find a better answer.

**Pull request merged**